render-ws-java-client/src/main/java/org/janelia/render/client/newsolver/solvers/ADDIntensityCorrectionClient.java

```diff
@@ -0,0 +1,112 @@
+package org.janelia.render.client.newsolver.solvers;
+
+import org.janelia.render.client.ClientRunner;
+import org.janelia.render.client.RenderDataClient;
+import org.janelia.render.client.newsolver.DistributedIntensityCorrectionSolver;
+import org.janelia.render.client.newsolver.setup.IntensityCorrectionSetup;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Alternating Domain Decomposition solver:
+ * A solver that alternates between solving for a given block layout and a shifted one so that
+ * all tiles are placed in the center of a block in one of the iterations.
+ * See also: additive Schwarz domain decomposition.
+ *
+ * @author Michael Innerberger
+ */
+public class ADDIntensityCorrectionClient {
+
+	public static void main(String[] args) {
+		if (args.length == 0) {
+			args = new String[]{"--baseDataUrl", "http://renderer-dev.int.janelia.org:8080/render-ws/v1",
+					"--owner", "hess_wafer_53",
+					"--project", "cut_000_to_009",
+					"--stack", "c009_s310_v01_mfov_08_test",
+					"--targetStack", "c009_s310_v01_mfov_08_test_ic",
+					"--minX", "33400",
+					"--maxX", "54600",
+					"--minY", "400",
+					"--maxY", "18700",
+					"--minZ", "424",
+					"--maxZ", "460",
+
+					"--blockSizeX", "7000",
+					"--blockSizeY", "6000",
+
+					"--completeTargetStack",
+
+					"--threadsWorker", "1",
+					"--threadsGlobal", "5",
+
+					"--zDistance", "0",
+
+					"--alternatingRuns", "4"
+			};
+		}
+
+		final ClientRunner clientRunner = new ClientRunner(args) {
+			@Override
+			public void runClient(final String[] args) throws Exception {
+
+				final IntensityCorrectionSetup parameters = new IntensityCorrectionSetup();
+				parameters.parse(args);
+
+				LOG.info("runClient: entry, parameters={}", parameters);
+
+				final ADDIntensityCorrectionClient client = new ADDIntensityCorrectionClient();
+
+				client.solveAlternating(parameters);
+			}
+		};
+		clientRunner.run();
+	}
+
+	private void solveAlternating(final IntensityCorrectionSetup parameters) throws IOException {
+		final String targetStackName = parameters.targetStack.stack;
+		final int nRuns = parameters.alternatingRuns.nRuns;
+
+		for (int runNumber = 1; runNumber <= nRuns; runNumber++) {
+			LOG.info("solveAlternating: run {} of {}; parameters={}", runNumber, nRuns, parameters);
+
+			parameters.targetStack.stack = getStackName(targetStackName, runNumber, nRuns);
+			DistributedIntensityCorrectionSolver.run(parameters);
+
+			if ((! parameters.alternatingRuns.keepIntermediateStacks) && (runNumber > 1))
```

Comment:
Minor nit, but something we should discuss at least once ... I prefer including brackets for all if statements (ostensibly to prevent accidental errors later, but really it is just a habit for me).  Both the [Oracle code conventions](https://www.oracle.com/java/technologies/javase/codeconventions-statements.html#:~:text=Braces%20are%20used%20around%20all,to%20forgetting%20to%20add%20braces.) and the [Google style guide](https://google.github.io/styleguide/javaguide.html#:~:text=Braces%20are%20used%20with%20if,a%20lambda%20expression%2C%20remain%20optional.) follow the same pattern.  Hoping I can convince you to do the same.

Reply:
I'm usually trying to save some lines by not including brackets. From my perspective, the indentation alone signals that there is a block (like in Python). The IDE warns if there is an indented line (i.e., should belong to statement) after the first without brackets, so I don't really agree with the error-proneness argument.

Also, I don't care too much and I'm happy to follow best practices (also thanks for pointing me to those; I was looking for something like that but somehow didn't find the Oracle conventions). :)